client/ayon_core/hosts/hiero/plugins/publish/collect_workfile.py

```diff
@@ -0,0 +1,39 @@
+import os
+
+import pyblish.api
+
+import hiero
+import nuke
+
+
+class CollectWorkfile(pyblish.api.InstancePlugin):
+    """Collect the current working file into context"""
+
+    families = ["workfile"]
+    label = "Collect Workfile"
+    order = pyblish.api.CollectorOrder - 0.49
+
+    def process(self, instance):
+        current_file = os.path.normpath(hiero.ui.activeProject().path())
+
+        # creating instances per write node
+        staging_dir = os.path.dirname(current_file)
+        base_name = os.path.basename(current_file)
+
+        # creating representation
+        representation = {
+            'name': 'nk',
+            'ext': 'nk',
+            'files': base_name,
+            "stagingDir": staging_dir,
+        }
+
+        # creating instance data
+        instance.data.update({
+            "name": base_name,
+            "representations": [representation]
+        })
```

Comment:
```suggestion
```

Reply:
This is about the Nuke host, not Hiero, so it shouldn't be used here.

The collector's job is to grab the .hrox workfile path and put it in [[instance.context.data](http://instance.context.data/)](http://instance.context.data)["currentFile"]. This info is then used in **extract_workfile.py**. I've seen that some of this is done in **collect_otio_timeline.py**. My idea is to take that part out and move it to **collect_workfile.py**. The Collect OTIO plugin should just change the active timeline into an otio_timeline and add it to the context data. Make sure everything is in the right order, please.

Reply:
i've updated the collector now to only collect current file.

It appears the Nuke host collect workfile directly collects the representation data, because there is no extract workfile.